source/bot/bot.py

```diff
@@ -1,10 +1,77 @@
 from addressbook import AddressBook
+from addressbook.records.record import Record
+from datetime import datetime
+from collections import defaultdict
+
 
 class Bot:
     def __init__(self) -> None:
         self.book = AddressBook()
         pass
 
     def idle(self) -> None:
-        # TODO: Implemennt main loop of a bot
-        pass
+        while True:
```

Comment:
It's better to have all commands in a separate functions, so we can call them easily when we need.